cds/modules/deposit/links.py

```diff
@@ -46,3 +47,34 @@ def deposit_links_factory(pid):
     )
 
     return links
+
+
+def cdsdeposit_links_factory(pid, deposit_type):
+    """Factory for links generation."""
+    def _url(name, **kwargs):
+        """URL builder."""
+        endpoint = '.{0}_{1}'.format(deposit_type, name)
+        return url_for(endpoint, pid_value=pid.pid_value, _external=True,
+                       **kwargs)
+    links = {}
+    links['self'] = _url('item')
+    links['files'] = _url('files')
+    ui_endpoint = current_app.config.get('DEPOSIT_UI_ENDPOINT')
+    if ui_endpoint is not None:
+        links['html'] = ui_endpoint.format(
+            host=request.host,
+            scheme=request.scheme,
+            pid_value=pid.pid_value,
+        )
+
+    for action in ('publish', 'edit', 'discard'):
+        links[action] = _url('actions', action=action)
+    return links
```

Comment:
@hachreak Please include also the ```bucket``` :)

```python
links['bucket'] = default_bucket_link_factory(pid)
```

https://github.com/CERNDocumentServer/cds/blob/603fa1d5216451b8db03bda480f09fc468ac507d/cds/modules/deposit/links.py#L38